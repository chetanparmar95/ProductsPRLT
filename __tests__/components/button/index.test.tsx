
import React from 'react';
import { render } from '@testing-library/react-native';
import Button from '../../../src/components/button'

describe('Button Component', () => {
  it('Button renders correctly', () => {
    const { toJSON } = render(<Button title="Click me" titleStyle={{color: 'black'}} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders button with provided title', () => {
    const { getByText } = render(<Button title="Click me" />);
    const buttonElement = getByText('Click me');
    expect(buttonElement).toBeDefined();
  });
})
