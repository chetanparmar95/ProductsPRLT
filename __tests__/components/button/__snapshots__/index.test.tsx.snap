// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FF5722",
      "borderRadius": 20,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontWeight": "bold",
          "marginLeft": 5,
        },
        {
          "color": "black",
        },
      ]
    }
  >
    Click me
  </Text>
</View>
`;
